### PR TITLE
utilities to pull 360 frames from the API

### PR DESCRIPTION
**Pull request by anenglishgoat**
`sb.frames` returns the json as-is from the API, or a dataframe with columns `event_uuid`, `visible_area` (list of vertices), `freeze_frame` (dict).

`sb.competition_frames` pulls all 360 frames for a comp and returns either a list of dicts (one per match) or a dataframe with all FFs.

if the wrong creds are supplied `frames` returns an empty dict/dataframe with a warning.

mirrored the tests for other functions and applied black formatting/removed some unused imports/generally cleaned up some stuff too.

### CODE REVIEW

**Comment by dvatvani** [comment]
* Tweaked the frames function to return a row per player-frame.
* Added examples of how to use the new functions in the README file

File: README.md [hunk 1]
```diff
@@ -1206,4 +1283,15 @@ sb.competition_events(
     gender="male",
     fmt="dict"
 )
+
+sb.frames(303299, fmt="dict")
```
Comment (WillTGM) [inline README.md:1287]: I don't think that `match_id` has 360 frames?

File: README.md [hunk 2]
```diff
@@ -1206,4 +1283,15 @@ sb.competition_events(
...
+
+sb.competition_frames(
+    country="Germany",
+    division= "1. Bundesliga",
+    season="2019/2020",
+    gender="male",
+    fmt="dict"
+)
```
Comment (WillTGM) [inline README.md:1295]: That season doesn't have 360 frames yet, so probably not a good example?

File: statsbombpy/sb.py [hunk 3]
```diff
@@ -121,3 +121,52 @@ def competition_events(
...
+) -> Union[pd.DataFrame, dict]:
+    if api_client.has_auth(creds) is True:
+        frames = api_client.frames(match_id, creds=creds)
+    else:
+        frames = public.frames(match_id)
+    if fmt == "dataframe":
+        frames = pd.json_normalize(
+            frames, "freeze_frame", ["event_uuid", "visible_area"]
+        )
+        frames = frames.rename(columns={"event_uuid": "id"})
```
Comment (WillTGM) [inline statsbombpy/sb.py:139]: Are the `id` fields consistent across the frames and events i.e. they are both `int` or `str`, so can be joined on `id`?

**Review by WillTGM — APPROVED** [review]
Looks good to me. Left a couple of comments re. the examples as those competitions don't have 360 data I believe.

File: statsbombpy/sb.py (hunk 3 above)
Comment (dvatvani) [reply statsbombpy/sb.py:139]: Yep

**Pull request merged**